integration/ztoc_test.go

```diff
@@ -145,3 +166,227 @@ func ztocExistChecker(t *testing.T, listOutputLines []string, img testImageIndex
+
+			}
+			if zinfoFile.Offset != int64(ztocFile.UncompressedOffset) {
+				t.Fatalf("different file offset: expected %v got %v", int64(ztocFile.UncompressedOffset), zinfoFile.Offset)
+
+			}
+			if zinfoFile.Size != int64(ztocFile.UncompressedSize) {
+				t.Fatalf("different file size: expected %v got %v", int64(ztocFile.UncompressedSize), zinfoFile.Size)
+
+			}
+		}
+	}
+
+	for _, img := range testImages {
+		img := img
+		tests := []struct {
+			name       string
+			ztocDigest string
+			expectErr  bool
+		}{
+			{
+				name:       "Empty ztoc digest",
+				ztocDigest: "",
+				expectErr:  true,
+			},
+			{
+				name:       "Invalid ztoc digest format",
+				ztocDigest: "hello",
+				expectErr:  true,
+			},
+			{
+				name:       "Invalid ztoc digest length",
+				ztocDigest: "sha256:hello",
+				expectErr:  true,
+			},
+			{
+				name:       "Ztoc digest does not exist",
+				ztocDigest: testutil.RandomDigest(),
+				expectErr:  true,
+			},
+			{
+				name:       "Correct ztoc digest",
+				ztocDigest: img.ztocDigests[0],
+				expectErr:  false,
+			},
+		}
+		for _, tt := range tests {
+			t.Run(tt.name, func(t *testing.T) {
+				var zinfo Info
+				output, err := sh.OLog("soci", "ztoc", "info", tt.ztocDigest)
+				if !tt.expectErr {
+					err := json.Unmarshal(output, &zinfo)
+					if err != nil {
+						t.Fatalf("expected Info type got %s", output)
+					}
+					ztocPath := filepath.Join(blobStorePath, trimSha256Prefix(tt.ztocDigest))
+					ztoc, err := getFullZtoc(sh, ztocPath)
+					if err != nil {
+						t.Fatalf("failed getting original ztoc")
+					}
+					verifyInfoOutput(t, zinfo, ztoc)
+				} else if err == nil {
+					t.Fatal("failed to return error")
+				}
+			})
+		}
+	}
+
+}
+
+func TestSociZtocGetFile(t *testing.T) {
+	t.Parallel()
+	sh, done := newSnapshotterBaseShell(t)
+	defer done()
+	rebootContainerd(t, sh, "", "")
+
+	testImages := prepareSociIndices(t, sh)
+
+	var (
+		tempOutputStream = "test.txt"
+		randomFile       = "ab2423n2j3h2jh32jn323"
+		randomZtocDigest = testutil.RandomDigest()
+	)
+
+	getFilePathWithinZtoc := func(t *testing.T, ztocDigest string) string {
+		var (
+			zinfo Info
+			path  string
+		)
+		output := sh.O("soci", "ztoc", "info", ztocDigest)
+		json.Unmarshal(output, &zinfo)
+		for _, file := range zinfo.Files {
+			if file.Size > 0 {
+				path = file.Filename
+			}
+		}
+		return path
+	}
+	verifyOutputStream := func(t *testing.T, sh *dockershell.Shell, contents, output []byte) {
+		if len(output) > 0 {
+			output = output[:len(output)-1]
+			res := bytes.Compare(contents, output)
+			if res != 0 {
+				t.Fatalf("output: %s expected: %s", output, contents)
+			}
+
+		} else {
+			o := sh.O("cat", tempOutputStream)
+			res := bytes.Compare(contents, o)
+			if res != 0 {
+				t.Fatalf("output: %s expected: %s", o, contents)
+			}
+
+		}
+	}
+
+	for _, img := range testImages {
+		img := img
+		ztocDigest := img.ztocDigests[0]
+		fileName := getFilePathWithinZtoc(t, ztocDigest)
```

Comment:
This would return the path for the first non-zero sized regular file. Sure

Reply:
@rdpsin This will be really IO intensive / take alot of time, should this only be for the stdout test and not the output test. I just ran this and it takes around 5 mins, just for stdout

Reply:
Can we do something like select a random subset of files (like a maximum of 10 files)? I'm afraid just testing a single file at the start/end can lead to false positives. Depends on whether we want the tests to be deterministic all the time.

Reply:
Yeah , I think that makes sense. And I do think the tests at least in this case shouldn't be deterministic.

Reply:
I do think we want them to be somewhat determinstic (or atleast catch regressions when and if they happen). FWIW, I don't think 5 mins is really all that time. :)

Reply:
5 mins does sound a little too long for running the test...A random subset of files should work.